src/main/java/org/assertj/core/internal/Classes.java

```diff
@@ -231,6 +263,115 @@ public void assertHasDeclaredFields(AssertionInfo info, Class<?> actual, String.
+    doAssertHasMethods(info, actual, actual.getDeclaredMethods(), methods);
+  }
+
+  private void doAssertHasMethods(AssertionInfo info, Class<?> actual, Method[] methods, String... expectedMethods) {
+    Set<String> expectedMethodNames = newLinkedHashSet(expectedMethods);
+    Set<String> missingMethodNames = newLinkedHashSet();
+    Set<String> actualMethodNames = methodsToName(methods);
+
+    if (!noMissingElement(actualMethodNames, expectedMethodNames, missingMethodNames)) {
+      throw failures.failure(info, shouldHaveMethods(actual, expectedMethodNames, missingMethodNames));
+    }
+  }
+
+  /**
+   * Verifies that the actual {@code Class} has the public {@code methods}.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the "actual" {@code Class}.
+   * @param methods the public methods who must be present in the class.
+   * @throws AssertionError if {@code actual} is {@code null}.
+   * @throws AssertionError if the actual {@code Class} doesn't contains all of the public methods.
+   */
+  public void assertHasPublicMethods(AssertionInfo info, Class<?> actual, String... methods) {
+    assertNotNull(info, actual);
+    doAssertHasPublicMethods(info, actual, actual.getMethods(), methods);
+  }
+
+  /**
+   * Verifies that the actual {@code Class} has the declared public {@code methods}.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the "actual" {@code Class}.
+   * @param methods the public methods who must be present in the class.
+   * @throws AssertionError if {@code actual} is {@code null}.
+   * @throws AssertionError if the actual {@code Class} doesn't contains all of the public methods.
+   */
+  public void assertHasDeclaredPublicMethods(AssertionInfo info, Class<?> actual, String... methods) {
+    assertNotNull(info, actual);
+    doAssertHasPublicMethods(info, actual, actual.getDeclaredMethods(), methods);
+  }
+
+  private void doAssertHasPublicMethods(AssertionInfo info, Class<?> actual, Method[] methods, String... expectedMethods) {
+    Set<String> expectedMethodNames = newLinkedHashSet(expectedMethods);
+    Set<String> missingMethodNames = newLinkedHashSet();
+    Map<String,Integer> actualMethods = methodsToNameAndModifier(methods);
+
+    if (!noMissingElement(actualMethods.keySet(), expectedMethodNames, missingMethodNames)) {
+      throw failures.failure(info, shouldHaveMethods(actual, expectedMethodNames, missingMethodNames));
+    }
+    Map<String,String> nonMatchingModifiers= new LinkedHashMap<>();
+    if(!noNonMatchingModifier(expectedMethodNames, actualMethods, nonMatchingModifiers, Modifier.PUBLIC)) {
+
+      throw failures.failure(info, shouldHaveMethods(actual, expectedMethodNames, Modifier.toString(Modifier.PUBLIC), nonMatchingModifiers));
+    }
+  }
+
+  private static boolean noNonMatchingModifier(Set<String> expectedMethodNames, Map<String, Integer> actualMethods, Map<String, String> nonMatchingModifiers, int modifier) {
```

Comment:
rename `actualMethods` to better express what this map contains (`methodsModifer` ?)